### DISCUSSION TITLE
node fs functions are undefined

### DISCUSSION THREAD

**Discussion by RohanG0407** [parcel-bundler/parcel#4790] [General]
# 🐛 bug report

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  {
  "name": "rohang0407.github.io",
  "version": "1.0.0",
  "description": "**This directory is not required, you can delete it if you don't want to use it.**",
  "main": "src/three/landingThree.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "deploy": "gh-pages -d dist",
    "dev": "parcel src/pages/index.html src/pages/about.html --open",
    "prod": "parcel build src/pages/index.html --public-url './' --no-content-hash && npm run deploy ",
    "build": "parcel build src/pages/index.html --public-url './' "
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/RohanG0407/myPortfolio.git"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/RohanG0407/myPortfolio/issues"
  },
  "homepage": "https://rohang0407.github.io/myPortfolio/",
  "dependencies": {
    "@barba/core": "^2.9.7",
    "gh-pages": "^3.0.0",
    "gsap": "^3.3.3",
    "jquery": "^3.5.1",
    "parcel-plugin-asset-copier": "^1.1.0",
    "three": "^0.117.1"
  },
  "devDependencies": {
    "parcel-bundler": "^1.12.4",
    "sass": "^1.26.8"
  },
  "assetsPath": "src/assets"
}

}
```

## 🤔 Expected Behavior

FS should read directory and print out filenames

## 😯 Current Behavior

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/24467483/85321964-8d2dc280-b493-11ea-975a-6858f1315c33.png">

## 💻 Code Sample

```
import fs from 'fs'
import gsap from 'gsap'

fs.readdir('./', function(err, items) {
    console.log(items);

    for (var i=0; i<items.length; i++) {
        console.log(items[i]);
    }
});

```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.12.4
| Node             | 13.11.0
| npm/Yarn         | 6.13.7
| Operating System | MACOS

**Comment by mischnic** [comment]
What are you trying to do here? There is no filesystem in the browser (at least by default)

**Comment by RohanG0407** [comment]
I’m trying to get the names of all the files in the dist folder

**Comment by mischnic** [comment]
This isn't possible with readDir, something like this might work for you:
```js
import assets from "./*.png";

// assets = { "a.png": "....", "b.png": ... }

**Comment by RohanG0407** [comment]
Well I'm essentially trying to get the names of the js files after building in the list folder with the hashes.

**Comment by mischnic** [comment]
Even if Parcel replaced `fs.readdir('./'` with a list of the folder contents at build time, it wouldn't work because these output files aren't there yet when that file is compiled.

I think you are mixing up server and client JS

**Comment by RohanG0407** [comment]
Is there anyway to get the names with hashes after building? Im essentially trying to inject my scripts so they can reload after transition but idk the names of the hashes so I have to change it every time. This is thing I need to do for barbjs transitions.

```
beforeEnter({next}) {
                console.log("about");
                let scss = document.createElement('link');
                let script = document.createElement('script');
                script.src = 'menu.1a1569bc.js';
                next.container.appendChild(script);
                scss.rel  = 'stylesheet';
                scss.href = '../css/about.scss';
                let scripts = document.getElementsByTagName('script');
                for(let i = 0; i < scripts.length; i++) {
                    console.log(scripts[i]);
                }
```

**Comment by mischnic** [comment]
So you need https://github.com/parcel-bundler/parcel/issues/4200?

**Comment by RohanG0407** [comment]
Yep, assuming its not yet implemented in Parcel 2?
